build.gradle

```diff
@@ -1531,15 +1531,15 @@ project(':test-common:test-common-runtime') {
   }
 
   dependencies {
-    implementation project(':test-common:test-common-internal-api')
-    implementation project(':clients')
-    implementation project(':core')
-    implementation project(':group-coordinator')
-    implementation project(':metadata')
-    implementation project(':raft')
-    implementation project(':server')
-    implementation project(':server-common')
-    implementation project(':storage')
+    api project(':core')
```

Comment:
Using "api" means consumers of ":test-common:test-common-runtime" don't have to re-declare all these deps.

Reply:
Is that a good idea though? Generally speaking, we should only declare `api` if the public classes in this module accept/return apis from the relevant module. Also, any module that uses classes from a given module should declare a dependency on that module (versus relying on transitive dependencies).

Reply:
We do need some of these as `api`. For example, ClusterInstance#brokers returns BrokerServer which is part of `:core`, so we need to declare that as an api dependency. 

I'll go through and see if we can keep some of these as `implementation`.